edb/pgsql/delta.py

```diff
@@ -1436,157 +1513,91 @@ def get_source_and_pointer_ctx(cls, schema, context):
-                alter_table_add_parent = source_ctx.op.get_alter_table(
-                    schema, context, force_new=True)
-
-                for base in old_base_order:
-                    parent_table_name = common.get_backend_name(
-                        schema, schema.get(base), catenate=False)
-                    cond = dbops.TableInherits(table_name, parent_table_name)
-                    op = dbops.AlterTableDropParent(
-                        parent_name=parent_table_name)
-                    alter_table_drop_parent.add_operation((op, [cond], None))
-
-                for added_base in new_base_order:
-                    parent_table_name = common.get_backend_name(
-                        schema, schema.get(added_base), catenate=False)
-                    cond = dbops.TableInherits(table_name, parent_table_name)
-                    op = dbops.AlterTableAddParent(
-                        parent_name=parent_table_name)
-                    alter_table_add_parent.add_operation((op, None, [cond]))
+    def schedule_inhviews_update(
+        self,
+        schema,
+        context,
+        obj,
+        *,
+        update_ancestors: Optional[bool]=None,
+        update_descendants: Optional[bool]=None,
+    ):
+        root = context.get(sd.DeltaRootContext).op
+        updates = root.update_inhviews.view_updates
+        update = updates.get(obj)
+        if update is None:
+            update = updates[obj] = InheritanceViewUpdate()
+
+        if update_ancestors is not None:
+            update.update_ancestors = update_ancestors
+        if update_descendants is not None:
+            update.update_descendants = update_descendants
+
+    def schedule_inhview_deletion(
+        self,
+        schema,
+        context,
+        obj,
+    ):
+        root = context.get(sd.DeltaRootContext).op
+        updates = root.update_inhviews.view_updates
+        updates.pop(obj, None)
+        deletions = root.update_inhviews.view_deletions
+        deletions[obj] = schema
+
+    def update_base_inhviews(self, schema, context, obj):
+        for base in obj.get_bases(schema).objects(schema):
+            self.schedule_inhviews_update(
+                schema, context, base, update_ancestors=True)
+
+    def update_lineage_inhviews(self, schema, context, obj):
+        self.schedule_inhviews_update(
+            schema, context, obj, update_ancestors=True)
+
+    def update_base_inhviews_on_rebase(
+        self,
+        schema,
+        orig_schema,
+        context,
+        obj,
+    ):
+        bases = set(obj.get_bases(schema).objects(schema))
+        orig_bases = set(obj.get_bases(orig_schema).objects(orig_schema))
+
+        for new_base in bases - orig_bases:
+            self.schedule_inhviews_update(
+                schema, context, new_base, update_ancestors=True)
+
+        for old_base in orig_bases - bases:
+            self.schedule_inhviews_update(
+                schema, context, old_base, update_ancestors=True)
```

Comment:
Wouldn't this sometimes update same ancestors multiple times?

Reply:
No, everything is eventually get collected into a set, so there are no repetitions.  You can schedule the view update on the same target as many times as you'd like.